packages/components/src/entities/actions.ts

```diff
@@ -0,0 +1,122 @@
+import { ActionURL, Ajax, Filter, Query, Utils } from '@labkey/api';
+
+import { SchemaQuery } from '../public/SchemaQuery';
+import { QueryInfo } from '../public/QueryInfo';
+import { getQueryDetails, selectDistinctRows, selectRowsDeprecated } from '../internal/query/api';
+import { downloadAttachment, findMissingValues } from '../internal/util/utils';
+import { getSampleTypeDetails } from '../internal/components/samples/actions';
+import { SCHEMAS } from '../internal/schemas';
+import { resolveErrorMessage } from '../internal/util/messaging';
+import { SAMPLE_ID_FIND_FIELD, UNIQUE_ID_FIND_FIELD } from '../internal/components/samples/constants';
+
```

Comment:
I think we should reconsider having this entire file. In general these actions files, in tandem with models files, are what leads to circular dependencies, and they are what makes our code hard to test. We have a few options:

1. Create an API Wrapper for these actions and inject the API wrapper where needed, this will make any consumer of these actions easier to test.
2. Move these actions where they are used.
    * downloadSampleType can be moved to SmapleTypeTemplateDownloadRenderer
    * getFindSamplesByIdData and getSamplesIdsNotFound can be moved to FindSamplesByIdPageBase
    * getSampleTypes should really live in an API Wrapper since it's not used in this repo. Or maybe it should live somewhere else entirely, or maybe there is an opportunity to move more shared code into this entrypoint that can be used in our apps.

Reply:
moved / split accordingly as per option 2